llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2318,7 +2321,11 @@ static bool isNonZeroRecurrence(const PHINode *PN) {
 
 static bool isNonZeroAdd(const APInt &DemandedElts, unsigned Depth,
                          const SimplifyQuery &Q, unsigned BitWidth, Value *X,
-                         Value *Y, bool NSW) {
+                         Value *Y, bool NSW, bool NUW) {
+  if (NUW)
+    return isKnownNonZero(Y, DemandedElts, Depth, Q) ||
+           isKnownNonZero(X, DemandedElts, Depth, Q);
```

Comment:
Separate refactor patch?

Reply:
This is pretty NFC. The callsites where all checking this manually before handle. Since we want NUW in `isAddNonZero` seemed to just make sense. But no strong feelings